src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            List<? extends Machine> bulldozers = bulldozerProducer.get();
+            return new ArrayList<>(bulldozers);
```

Comment:
```suggestion
            return new ArrayList<>(bulldozerProducer.get());
```
we can simplify each block